packages/mobile/src/escrow/ReclaimPaymentConfirmationScreen.tsx

```diff
@@ -58,13 +62,15 @@ const mapStateToProps = (state: RootState): StateProps => {
   }
 }
 
-type Props = DispatchProps &
-  StateProps &
-  WithTranslation &
-  StackScreenProps<StackParamList, Screens.ReclaimPaymentConfirmationScreen>
+type Props = DispatchProps & StateProps & WithTranslation & ScreenProps
 
 class ReclaimPaymentConfirmationScreen extends React.Component<Props> {
-  static navigationOptions = { header: null }
+  componentDidMount() {
+    this.props.navigation.addListener('beforeRemove', (e) => {
+      this.props.reclaimEscrowPaymentCancel()
```

Comment:
This is going to be fired even if the escrow payment succeeds, since we are still going back in that case. How about only running this two lines if `isReclaiming` is `true` in the store? Calling `reclaimEscrowPaymentCancel` in case of success looks harmless but the analytics event could be confusing.

Reply:
Good suggestion. Done.